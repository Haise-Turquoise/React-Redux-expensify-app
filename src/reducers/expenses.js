// Expensese Reducer

const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [...state, action.expense];//state.concat(action.expense);
        case 'REMOVE_EXPENSE':
            // console.log('REMOVE_EXPENSE');
            // console.log('each id: ', state);//.expenses
            // console.log('action.id: ', action.id);
            return state.filter(({ id }) => (id !== action.id));
            //return state.filter(({ id }) => (id !== action.id));
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id == action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    return expense;
                }
            });
        default:
            return state;
    }
};

export default expensesReducer;