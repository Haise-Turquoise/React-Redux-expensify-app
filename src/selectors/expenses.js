// Get visible expenses
import moment from 'moment';
const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    // console.log('sortBy: ', sortBy);
    return expenses.filter( (expense)=>{
        const startDateMatch = startDate ? startDate.isSameOrBefore(expense.createdAt, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(expense.createdAt, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    
        // console.log(expense.description, ': ');
        // console.log('expense.createdAt: ',expense.createdAt);
        // console.log('startDate: ',startDate);
        // console.log('startDateMatch: ',startDateMatch);
        // const endDateMatch = 
        // const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        // return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b)=> {
        switch (sortBy) {
            case 'date':
                return a.createdAt < b.createdAt ? 1 : -1;
            case 'amount':
                return a.amount < b.amount ? 1 : -1;
            // if (sortBy === 'date') {
            //     return a.createdAt < b.createdAt ? 1 : -1;
            // }
            // if (sortBy === 'amount') {
            //     return -(a.amount - b.amount);
            // }
        }
    });
};

export default getVisibleExpenses;