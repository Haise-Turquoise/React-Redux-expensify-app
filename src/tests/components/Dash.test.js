import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";
import DashPage from '../../components/Dash';
import NotFoundPage from '../../components/NotFound';
import moment from "moment";

test("should render DashPage with expenses", () => {
    const wrapper = shallow(<DashPage />);
    expect(wrapper).toMatchSnapshot();
    }
);