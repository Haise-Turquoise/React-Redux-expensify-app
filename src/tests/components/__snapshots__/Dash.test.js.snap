// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashPage with expenses 1`] = `
<div>
  <Connect(ExpenseListFilter) />
  <Connect(ExpenseList) />
</div>
`;
